targets/TARGET_STM/us_ticker_16b.c

```diff
@@ -25,7 +25,6 @@ TIM_HandleTypeDef TimMasterHandle;
 
 volatile uint32_t SlaveCounter = 0;
 volatile uint32_t oc_int_part = 0;
-volatile uint16_t oc_rem_part = 0;
 
 static int us_ticker_inited = 0;
 
```

Comment:
With your changes, set_compare() is now called from only a single place.  It might be better to just put the contents of this function inline at that one place.  You could then make the `TimMasterHandle.Instance = TIM_MST;` assignment once at the top of the us_ticker_set_interrupt() function, rather than separately in both the "if" and "else" cases.  You may go even further and just provide a static initializer for TimMasterHandle to set Instance, so it doesn't have to be done on every function call (e.g. `TIM_HandleTypeDef TimMasterHandle = { TIM_MST }; // Initialize Instance`).